### DISCUSSION TITLE
antlr4.13 target javascript or typescript generate python code in the lexer file

### DISCUSSION THREAD

**Discussion by Elmahdi1962** [antlr/antlr4#4315] [Q&A] [answered]
when using the javascript or typescript target to genearte the parser the lexer file has pythoncode combined with little bit of javascript code,
```
// Generated from src/backend/pythonAntlrParser/Python3.g4 by ANTLR 4.9.0-SNAPSHOT


from antlr4.Token import CommonToken
import re
import importlib

# Allow languages to extend the lexer and parser, by loading the parser dynamically
module_path = __name__[:-5]
language_name = __name__.split('.')[-1]
language_name = language_name[:-5]  # Remove Lexer from name
LanguageParser = getattr(importlib.import_module('{}Parser'.format(module_path)), '{}Parser'.format(language_name))


import { ATN } from "antlr4ts/atn/ATN";
import { ATNDeserializer } from "antlr4ts/atn/ATNDeserializer";
import { CharStream } from "antlr4ts/CharStream";
import { Lexer } from "antlr4ts/Lexer";
import { LexerATNSimulator } from "antlr4ts/atn/LexerATNSimulator";
import { NotNull } from "antlr4ts/Decorators";
import { Override } from "antlr4ts/Decorators";
import { RuleContext } from "antlr4ts/RuleContext";
import { Vocabulary } from "antlr4ts/Vocabulary";
import { VocabularyImpl } from "antlr4ts/VocabularyImpl";

import * as Utils from "antlr4ts/misc/Utils";


export class Python3Lexer extends Lexer {
	public static readonly STRING = 1;
	public static readonly NUMBER = 2;
	public static readonly INTEGER = 3;
	public static readonly DEF = 4;
	public static readonly RETURN = 5;
	public static readonly RAISE = 6;
```
	
	`line 193:	// @Override
	// @NotNull
	public get vocabulary(): Vocabulary {
		return Python3Lexer.VOCABULARY;
	}
	// tslint:enable:no-trailing-whitespace

	@property
	def tokens(self):
	    try:
	        return self._tokens
	    except AttributeError:
	        self._tokens = []
	        return self._tokens

	@property
	def indents(self):
	    try:
	        return self._indents
	    except AttributeError:
	        self._indents = []
	        return self._indents

	@property
	def opened(self):
	    try:
	        return self._opened
	    except AttributeError:
	        self._opened = 0
	        return self._opened`

**Comment by ericvergnaud** [comment, marked as answer]
The culprit code comes from the grammar not from antlr.
You need to use a grammar with production code suitable for JS

**Reply by Elmahdi1962** [reply]
i used the python3 grammar from the antlr's grammar repo as i want to parse python code https://github.com/antlr/grammars-v4/blob/master/python/python3-py/Python3.g4

**Reply by Elmahdi1962** [reply]
yea i apparently i took the wrong one
